hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlExecuteMessageTask.java

```diff
@@ -0,0 +1,124 @@
+import com.hazelcast.sql.impl.SqlResultImpl;
+import com.hazelcast.sql.impl.SqlServiceImpl;
+
+import java.security.Permission;
+
+/**
+ * SQL query execute task.
+ */
+public class SqlExecuteMessageTask extends AbstractCallableMessageTask<SqlExecuteCodec.RequestParameters> {
+    public SqlExecuteMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected Object call() throws Exception {
+        try {
+            SqlQuery query = new SqlQuery(parameters.sql);
+
+            if (parameters.parameters != null && !parameters.parameters.isEmpty()) {
+                for (Data param : parameters.parameters) {
+                    query.addParameter(serializationService.toObject(param));
+                }
+            }
+
+            query.setTimeoutMillis(parameters.timeoutMillis);
+            query.setCursorBufferSize(parameters.cursorBufferSize);
+
+            SqlServiceImpl sqlService = nodeEngine.getSqlService();
+
+            SqlResultImpl cursor = (SqlResultImpl) sqlService.query(query);
+
+            SqlPage page = sqlService.getInternalService().getClientStateRegistry().registerAndFetch(
+                endpoint.getUuid(),
+                cursor,
+                parameters.cursorBufferSize,
+                serializationService
+            );
+
+            return new SqlExecuteResponse(
+                cursor.getQueryId(),
+                cursor.getRowMetadata(),
+                page,
+                null
+            );
+        } catch (Exception e) {
+            SqlError error = SqlClientUtils.exceptionToClientError(e, nodeEngine.getLocalMember().getUuid());
+
+            return new SqlExecuteResponse(null, null, null, error);
+        }
+    }
+
+    @Override
+    protected SqlExecuteCodec.RequestParameters decodeClientMessage(ClientMessage clientMessage) {
+        return SqlExecuteCodec.decodeRequest(clientMessage);
+    }
+
+    @Override
+    protected ClientMessage encodeResponse(Object response) {
+        SqlExecuteResponse response0 = (SqlExecuteResponse) response;
+
+        return SqlExecuteCodec.encodeResponse(
+            response0.getQueryId(),
+            response0.getRowMetadata(),
+            response0.getPage(),
+            response0.getError()
+        );
+    }
+
+    @Override
+    public String getServiceName() {
+        return SqlInternalService.SERVICE_NAME;
+    }
+
+    @Override
+    public String getDistributedObjectName() {
+        return null;
+    }
+
+    @Override
+    public String getMethodName() {
+        return "execute";
+    }
+
+    @Override
+    public Object[] getParameters() {
+        return new Object[] {
+            parameters.sql,
+            parameters.parameters,
+            parameters.timeoutMillis,
+            parameters.cursorBufferSize
+        } ;
+    }
+
+    @Override
+    public Permission getRequiredPermission() {
+        return null;
```

Comment:
Is permission left `null` on purpose? I think, SQL permissions are important.

Reply:
We do not have a security model yet. Given that an SQL script could trigger any kind of action (e.g. read from several maps, or start Jet job, etc), it is impossible to define the required permissions on that level. The permissions will be known after parsing only.
Meanwhile, we may add a global flag to disable SQL on the member, but this is out of scope of this PR.

Reply:
I think we may define a global "query" permission that will govern whether the user can execute any query or not. And more deep integration will be implemented later.

Reply:
Added the `sql` permission, modeled after the `transaction` permission.